File: 2511 captureForts/2511.py
class Solution:
    def captureForts(self, forts: list[int]) -> int:
        ans=0
        pre=-1
        for i,fort in enumerate(forts):
            if fort==-1 or fort==1:
                if pre>=0 and fort!=forts[pre]:
                    ans=max(ans,i-pre-1)
                pre=i
        return ans

File: 2511 captureForts/2511.c
int captureForts(int* forts, int fortsSize){
    int ans=0,pre=-1;
    for(int i=0;i<fortsSize;i++){
        if(forts[i]==1 || forts[i]==-1){
            if(pre>=0 && forts[pre]!=forts[i]){
                ans=ans>(i-pre-1)?ans:(i-pre-1);
            }
            pre=i;
        }
    }
    return ans;
}
